### PR TITLE
Dialogs revamp

### PR DESCRIPTION
**Pull request by pusztaienike**
What this PR contains:

- new urls have been created for new, edit and browse (reused the editProperties)
- all the fields and react-control-mapper have been copied from sn-control-react package and have been restyled
- the old, not-used dialogs have been cleaned from the package
- new props have been used for the Explore component to catch data from ContentContextMenu. check this: [diagram](https://app.diagrams.net?lightbox=1&highlight=0000ff&edit=_blank&layers=1&nav=1&title=PropsChangesCauseOfForm.drawio#R7Vxdc9o4FP01TLsPZWzJn48Bkt2Hdrcz6XSbRwUEeGsjjywa0l%2B%2Fsi1jWzJgYrAolHYmtiR%2FcM85V1dXEgM4jjZ%2FUhQvP5EZDgfAmG0GcDIAwAS%2Bz%2F%2BkJa95iW94ecGCBjPRqCx4DH5iUWiI0nUww0mtISMkZEFcL5yS1QpPWa0MUUpe6s3mJKw%2FNUYLrBQ8TlGolv4bzNhSlJqGUVb8hYPFUjzas0XFM5p%2BX1CyXonnDQCcZ5%2B8OkLFvUT7ZIlm5KVSBO8HcEwJYflRtBnjMLVtYbb8uocdtdv3pnjF2lwAl7brfHky7u%2FjYDl6%2FPrhBdMPjng39lrYA8%2B4ecQpoWxJFmSFwvuydJR9Z5ze1eBnZZuPhMS80OSF%2F2HGXgXWaM0IL1qyKBS1eBOwb%2BLy9PgpPR7a4myyqVRNXsVJ%2Fp7py%2B38%2BqIoIWs6xXu%2Bc8EyRBeY7WkHtiBx8mMSYUZf%2BXUUh4gFP%2BrvgQQLF9t2JRL8QIBxBDCuTmDMI4Dh5qev38qG6elTta68LDvTCKilE1BPq9K2ID5Van55QKFOQMV9f6BwLZ7EuyiWftfUET%2Bkl3PwNkOWbBToS2BTLF6WAcOPMcps8sK72TqI4jmYMrzZb1rVFOICy7LzS0QnbYk%2B6aXS4xX91LLS2TnGmYwHgBY5lNQG0K2Re2hAeIDg2dlnTANuAkxF4ZxjLp4H0vMQPeNwtA0NxiQkadMVWWHRuiji8YJh%2BPzDyxNGyfdt%2FGGlLYMwrLScIezNp9uWlRpn6uHn%2BWn1B1rqz%2B%2Bov%2BzSO0rRa6VBTIIVSyp3%2FpwWlFyGvlvjsukaEh3zO5bk3L5aB74qYh%2BTKEKcjpQfxBzd9BV1q9yoG6ZJ5U6vKtfS6e2XYF1qZoOEd4n0hAKDLQVmGs2An1dhNqgTidugSgWlvb%2B%2F%2FXkUCc1L55Z1BLe0uvu2bOzq7jv5Eqh44PtNzM2C9btdKbhq9Lv9Rle%2B5ujKPWrkcJWisn4FUVmKqB6DKA7xxyDRH864fot4BvapK6hl1NKXPGzszawmeXjgGTrOaeXht5RH1yH%2BmwIgS2IecPcHQAfanykA0pxRAm9KKVmWc0zX0DjmPnGA77sTw3U10dvRwW9o1vlqmfv5rbR39rc3i4Ffc%2Fsz6UFL0HNFXDS7Tni8bbBp2Edxq6jvlVtF5KHL17rnT9%2BfOMrYEllrEN6a%2BlqnhnwlCh%2FnMwkXEYZDoy44aKtheNGB9BOG25rVWIt8jLZq9GyvqkdzaJjuAU12jn66CLXX4UDBn8scD3hWPZ4pMu27%2Bih7f%2Fvz9FGFCStu5AvFnApsOUEMafcjrmfVjeKofgTYTo9%2BxLL0%2BpH2buQ6NQ9aav4k04xqIAmlQNKSeJa%2Fl7iqpNrRgy3vuMGTBTQEuAUWagwyohjNxnQdPSeXM9NZJFgKG9lmgy%2FpNeVuaZmO2jFCaB2T%2BE49GzP0vTetgrhO%2F%2FRLpPBtqJN4b1wpWJ8fMg9Q7rbZZbk66eVoHmu9IfFhDmoureTab3o1NbS10uuCvJfb1n3V%2BQUuhV%2B9Ztba86vrDEe3RbrqDLeIa7M%2FG%2FYJr9baw1nbahHO9rpyz9az%2F2BH7vo2AwS7rcK6ztt0Y8rFT7sdg3h1hfhRFNTrjFtTpes8x458p5TisP12qZSTZS9sxctnaYvphactHKg7beHoSYFWZkV8UJPZ0DgUTt1QAsLpS9bdYixHUd98vZqygKwG8C59jfXqLjt9%2F4fCtmSJ4vSQ2weFIQ7JgqKImzGuAFyrqyB%2FSLESeHM7%2Fdfok7OPSoP8oxAInMYbbHcSF97AbfAGva5vBKofDZJ%2FYrx6oCRKz93RuyRbIPpu4E4ULNN4mpsGhcGCIz%2BZcjtkMKVWCqYovBMVUTCb5Y4EJ8FP9Jxdn6pQpNL5zezRwJ6k9%2BK%2BI8l1akrg7t2l0QkYx6rPVJmWrwCzXYxaBQacDRhVYoMxGNwZCjzTYluSJoTOCAuQVqE1wQKaFiKcDRaobkrYAQvO9ypcHyi2Ic3qNqwO6VcrUB3yJ5g9EBqlmLxHWU%2F0N4qw2hn1joS03K%2Fj7j9pEbLfgERTEuF8SKhe62tA2Zrb9yeefcnspjskbzGR2O8eAwsoRluicP5Z%2FIyJO8o9TExJnAwrfiYrTVvk%2F%2FNT4KAozgwFoZ8lNMWWKan8stzSibdnudIC06Ywy%2B5TF5babdyEh3LlZeQNw99%2BPZSlBrw3gQSQF1F72pHQs5vld8JZnmjuY1d6t4Qz0MoUaXb5Nmf%2F2lNFa2aqeE1pSaz%2BoFNaCgsb3G%2FjUtiivzq9pdQ9CDfREfrSIlDY9jdTztYR2urvSt0EEq48fNWPhJpIkEZi3AFSXnXFgyevTaBo9IqKmlS4CX340uDJbkq09asP7zaRkAdPjqEbCedG%2BwxbnkHTnvJ0Dmbvrr%2FPkKcEHKC7z3D2p3muTBWuvMqkIeHZs%2F3VPnurCu2mP1fk5IE6Cg0zCydyTfy0%2FBnxfE1V%2BVvt8P5%2F)
- double click on user function has been fixed

Action list for the future:

**TOP PRIO:**
- [ ] Restyle the version history view
- [ WIP ] Restlye the Binary editor view (Title + buttons should be FAB) #608 
- [ ] Decide what to do if field is empty in Browse view (Now only the label is visible)
- [ ] Check the old (small) dialogs (are-you-sure, check-in, etc.). Do we have design for them?
- [ ] reference fields' on browse view

**POSTPONED:**
- [x] Add default values for all the fields which are date/time (in CTDs)
- [ ] Handle if there is no default value for time or date use moment.now() function
- [ ] Format date-s on browse view
- [x] In calendar content Eventtype can be more than one option (allowedMultiple in CTDs) 
- [x] Generate the default schema from the new CTDs.
- [x] Write test for these changes (it will be e2e test)
- [x] icon on the edit forms (at the title) should be a link to the related ctd
- [x] shown fields and their order should be revisited on types: File, User
- [x] versions dialog should be revamped
- [ ] size field could be improved by displaying units (kB, MB)
- [x] set demo contents' validfrom and validtill fields' value with a valid date
- [ ] number field on browse view with value 0

### CODE REVIEW

File: apps/sensenet/src/components/new/NewProperties.tsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        oDataOptions={{ select: 'all' }}>
+        <CurrentAncestorsProvider>
+          <div className={clsx(classes.breadcrumbsWrapper, globalClasses.centeredVertical)}>
+            <ContentBreadcrumbs />
+          </div>
+          <NewView
+            handleCancel={history.goBack}
+            repository={repo}
+            contentTypeName={contentTypeName}
+            onSubmit={async content => {
```
Comment (B3zo0) [inline apps/sensenet/src/components/new/NewProperties.tsx]: If possible please move this logic to the NewView component.

Comment (pusztaienike) [reply apps/sensenet/src/components/new/NewProperties.tsx]: I tried, but for onSubmit we need the currentContextProvider

**Comment by herflis** [comment]
something is not okay on files' browse view with the locked field
image
Btw I think locked should be hidden from browse view :)

**Comment by herflis** [comment]
image

**Comment by herflis** [comment]
image

add content button is disabled after I opened a document to edit and navigated back to the parent folder with the breadcrumb

**Comment by herflis** [comment]
image
user browse view is a bit messy :)

File: apps/sensenet/src/components/view-controls/edit-view.tsx [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+    return (
+      <>
+        <form
+          className={clsx(classes.form, {
+            [classes.fullpage]: props.isFullPage,
+          })}
+          onSubmit={handleSubmit}>
+          <Grid container={true} spacing={2}>
+            {schema.fieldMappings
+              .sort((item1, item2) => item2.fieldSettings.DefaultOrder! - item1.fieldSettings.DefaultOrder!)
```
Comment (herflis) [inline apps/sensenet/src/components/view-controls/edit-view.tsx:159]:
```suggestion
              .sort((item1, item2) => item2.fieldSettings.FieldIndex! - item1.fieldSettings.FieldIndex!)
```

File: apps/sensenet/src/components/Icon.tsx [hunk 3]
```diff
@@ -284,12 +335,9 @@ export const IconComponent: React.FunctionComponent<{
   const repo = useRepository()
 
   const options: IconOptions = { style: props.style, injector, repo }
-  const resolvers = [
-    ...(props.resolvers || []),
-    ...defaultContentResolvers,
-    ...defaultSchemaResolvers,
-    ...defaultNotificationResolvers,
-  ]
+  const resolvers = props.resolvers
```
Comment (B3zo0) [inline apps/sensenet/src/components/Icon.tsx:338]: I like the original solution better. use the prop. resolver first and if it fails fall back to the defaults. Can you please revert this back and write a customResolver just for the File type?

**Pull request merged**